### PR TITLE
Implement user equality

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: lib/logic/models/user.js [hunk 1]
```diff
@@ -17,4 +17,11 @@ User.prototype.getUserLinkJsx = function(key) {
     );
 };
 
+User.prototype.equals = function(other) {
+    if (!other) {
+        return false;
+    }
+    return this.name == other.name;
```
Comment (kegsay) [inline lib/logic/models/user.js]: Triple `=` please, we have no reason to allow type coercion here.

File: lib/logic/list-controller.js [hunk 2]
```diff
@@ -92,14 +92,14 @@ ListController.prototype._loadAssignedPulls = function(prs) {
...
+        if (pr.getOwner().equals(me)) {
             // PR will appear in my pulls, don't duplicate it in assigned pulls
             continue;
         }
-        else if (pr.getLastCommenter() === pr.getOwner()) {
+        else if (pr.getOwner().equals(pr.getLastCommenter())) {
             assignedPulls.assignedNeedsAction.push(pr);
         }
-        else if (pr.getLastCommenter() !== me) {
+        else if (!me.equals(pr.getLastCommenter())) {
```
Comment (kegsay) [inline lib/logic/list-controller.js:102]: Do you know for 100% that `PullRequest.getAssignee()` and `PullRequest.getOwner()` won't ever return `null`? The latter I guess _maybe_ assuming GH APIs aren't being a crank, but the former surely not. Null checks please!

**Comment by kegsay** [comment]
LGTM other than comments.

File: lib/logic/models/user.js (hunk 1 above)
Comment (illicitonion) [inline lib/logic/models/user.js]: Done

File: lib/logic/list-controller.js (hunk 2 above)
Comment (illicitonion) [inline lib/logic/list-controller.js:102]: Yes; we're only looking at assigned pull requests here, so we must the the assignee. And all pull requests must have an owner. Unless github is buggy.

Comment (kegsay) [inline lib/logic/list-controller.js:102]: test message

Comment (kegsay) [inline lib/logic/list-controller.js:102]: test

Comment (kegsay) [inline lib/logic/list-controller.js:102]: test2

Comment (kegsay) [inline lib/logic/list-controller.js:102]: test3

Comment (kegsay) [inline lib/logic/list-controller.js:102]: done

**Pull request merged**